lib/ramfs/README.md

```diff
@@ -0,0 +1,95 @@
+# RamFS: Unikraft’s RAM-based Virtual Filesystem
+
+The `RamFS` filesystem is a virtual filesystem that uses memory for storage, as the name implies.
+Its advantage is that it is very fast.
+From a simplified perspective, we can look at a file in `RamFS` as a buffer in memory.
+Since this is how `RamFS` represents data, the disadvantage becomes apparent: we can only write data into `RamFS` until we fill up all the memory.
+Moreover, data is not persistent: once we stop the unikernel, memory is freed and data is lost.
+
+## Key Functions and Data Structures
+
+The key structure used is `ramfs_node` defined as:
+
+```c
+struct ramfs_node {
+    /* Next node in the same directory */
+    struct ramfs_node *rn_next;
+    /* First child node */
+    struct ramfs_node *rn_child;
+    /* Type: file or directory */
+    int rn_type;
+    /* Name of the file/directory (null-terminated) */
+    char *rn_name;
+    /* Length of the name, not including the terminator */
+    size_t rn_namelen;
+    /* Size of the file */
+    size_t rn_size;
+    /* Buffer to the file data */
+    char *rn_buf;
+    /* Size of the allocated buffer */
+    size_t rn_bufsize;
+    /* Last change time */
+    struct timespec rn_ctime;
+    /* Last access time */
+    struct timespec rn_atime;
+    /* Last modification time */
+    struct timespec rn_mtime;
+    /* Node access mode */
+    int rn_mode;
+    /* Whether or not the rn_buf was allocated in this ramfs_node */
+    bool rn_owns_buf;
+};
+```
+
+This structure contains the file type (which can be a regular file - `VREG`, a symbolic link - `VLNK`, or a directory - `VDIR`), the buffer in which the data will be stored, and its size.
+A field that normally should not be here is the name, but for the simplicity of the library, it is used.
+Unfortunately, the fact that the name field is here and is used in the code does not allow the creation of hard links.
+
+## Configuring Applications to Use `RamFS`
+
+We have the following simple app that we run using a `RamFS` filesystem:
+
+```c
+char buf[1024];
+int f = open("test1.txt", O_RDWR | O_CREAT, 0777);
```

Comment:
Use a less boring name for the file. Use something like `atlantis.txt` or `westeros.txt` or `server.conf`. And also the message could be something different than `TEST`.